File: wip/Euler12.py
# Project Euler
# Problem 12 - Highly divisible triangular number
# https://projecteuler.net/problem=12
# Created by Andrew Voss

currentRunNum = 0

def findFactors(n):
    return set(reduce(list.__add__, ([i, n//i] for i in range(1, int(n**0.5) + 1) if n % i == 0))) # factors

def valuesOfTriangleNums():
    global currentRunNum
    x = currentRunNum
    total = 0
    for i in range(1,x+1):
        total = total + i
        i = i + 1
    currentRunNum += 1
    return total

def main():
    run = 1
    e = 1
    while run:
        x = valuesOfTriangleNums()
        y = findFactors(x)
        if y > 5:
            run = 0
        e += 1;
        print x, y
    print x, y

print findFactors(224)

File: wip/Euler12.cpp
/*
 * Project Euler
 * Problem 12 - Highly divisible triangular number
 * https://projecteuler.net/problem=12
 * Created by Andrew Voss
 */

#include <iostream>
using namespace std;

int findFactors(int x){
  int value = x;
  int i;
  int numberOfDivisor = 0;

  for(i = 1; i <= value; i++){
    if(value % i == 0){
      numberOfDivisor++;
    }
  }
  return numberOfDivisor;
}

int triangleNum(int x){
  int num = 0;
  int i;
  for(i = 0; i <= x; i++){
    num = num + i;
  }
  return num;
}

int main()
{
  /*int x, y, i;
  for(int run = 1; run; i++){
    x = triangleNum(i);
    y = findFactors(x);
    if(y > 500){
      run = 0;
    }
  }

  cout << x << "\n" << y << endl;*/

  cout << triangleNum(10000000) << endl;
  cout << findFactors(16476686400) << endl;
}
